### PR TITLE
doc: add docs for deployment and configuration

### PR DESCRIPTION
**Pull request by darahayes**
### Description

I've added the docs but I don't really know if I need to link them up in another place to make them visible on the website. @pwright could I get some help on that?

Also, please ignore the failing build, we're still working to fix that.

##### Checklist

- [ ] `npm test` passes
- [ ] tests are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](../resources/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+** Using https://webpack.js.org/guides/getting-started/[Webpack Getting Started Guide]
+
+= Procedure
+
+. Get the hostname of the {sync-service} Server application.
+.. Run the command
++
+[source,bash]
+----
+oc get route <data-sync-application-name>
```
Comment (pwright) [inline docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc]: is that the namespace?

File: docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+[source,bash]
+----
+NAME                             HOST/PORT               PATH      SERVICES        PORT      TERMINATION   WILDCARD
+<sync-server-application-name>   <sync-server-hostname>            data-sync-app   <all>                   None
+----
++
+.. Record the value for `<sync-server-hostname>`.
+. Add the following libraries to your project:
++
+[source,javascript]
```
Comment (pwright) [inline docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc]: haven't they already run the app locally and therefore have these libraries?

File: docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+const config = {
+  httpUrl: 'https://<sync-server-hostname>/graphql',
+  wsUrl: 'wss://<sync-server-hostname>/graphql'
+}
+
+const client = await createClient(config)
+----
+
+NOTE: if your {sync-service} server application is not available through HTTPS, then the config should be changed as follows.
```
Comment (pwright) [inline docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc]: Is that a valid 'enterprise' use case?
suggestion: let's not support users using http :)

File: docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+. Add the following libraries to your project:
++
+[source,javascript]
+----
+npm install @aerogear/voyager-client
+npm install graphql
+npm install graphql-tag
+----
++
+. In your project source code, import and configure the client using the server hostname.
```
Comment (pwright) [inline docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc]: wondering: isn't it more typical that user would be modifying their existing local server, than doing this for first time?

File: docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc (hunk 2 above)
Comment (darahayes) [reply docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc]: I can change it to something like `ensure the following libraries are added to your project`.

File: docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc (hunk 3 above)
Comment (darahayes) [reply docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc]: I agree however the data sync template deploys things with http right now :(

File: docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc (hunk 4 above)
Comment (darahayes) [reply docs/integreatly/modules/ROOT/pages/ds-connect-client-to-server.adoc]: Do you mean modifying their existing client app? That is a fair point. I can change it to something like `if you haven't done so already, import and configure the client using the server hostname`

**Pull request merged**